source/Halibut/Transport/SecureClient.cs

```diff
@@ -47,7 +45,11 @@ public void ExecuteTransaction(Action<MessageExchangeProtocol> protocolHandler,
             var watch = Stopwatch.StartNew();
             for (var i = 0; i < ServiceEndpoint.RetryCountLimit && retryAllowed && watch.Elapsed < ServiceEndpoint.ConnectionErrorRetryTimeout; i++)
             {
-                if (i > 0) log.Write(EventType.Error, "Retry attempt {0}", i);
+                if (i > 0)
+                {
+                    Thread.Sleep(retryInterval);
+                    log.Write(EventType.OpeningNewConnection, $"Retrying connection to {ServiceEndpoint.Format()} - attempt #{i}.");
```

Comment:
nitpick: string interpolation here prevents structured logging.

Reply:
Agreed. We have no structured logging here, though - it's all string interpolation :( We need us a Serilog, we do.